lib/puppet_x/voxpupuli/splunk/util.rb

```diff
@@ -0,0 +1,32 @@
+require 'openssl'
+require 'base64'
+
+module PuppetX
+  module Voxpupuli
+    module Splunk
+      class Util
+        def self.decrypt(secrets_file, value)
+          return value unless value.start_with?('$7$')
+
+          Puppet.debug "Decrypting splunk >= 7.2 data using secret from #{secrets_file}"
+          value.slice!(0, 3)
+          data = Base64.strict_decode64(value)
+          splunk_secret = IO.binread(secrets_file).chomp
+
+          iv         = data.bytes[0, 16].pack('c*')
+          tag        = data.bytes[-16..-1].pack('c*')
+          ciphertext = data.bytes[16..-17].pack('c*')
+
+          decipher = OpenSSL::Cipher::AES.new(256, :GCM).decrypt
+          decipher.key = OpenSSL::KDF.pbkdf2_hmac(splunk_secret, salt: 'disk-encryption', iterations: 1, length: 32, hash: 'sha256')
+          decipher.iv_len = 16
+          decipher.iv = iv
+          decipher.auth_tag = tag
+          decipher.auth_data = ''
+
+          decipher.update(ciphertext) + decipher.final
```

Comment:
Dunno if I should catch exceptions here, log a warning, and return something so that puppet can continue??

Reply:
IMHO things this deep shouldn't rescue exceptions because they're too deep to know what to do about them. That should be done higher up. IMHO only rescue exceptions here if you watch to toss your own customized exception like `Spunk::Util::DecodeException` blah blah blah IMHO YMMV

Reply:
The thought was if decryption did fail, it could be because of something like the splunk.secrets file being replaced.  If decryption does fail (for whatever reason), it's maybe better for `insync?` to return false instead of a hard error being thrown and not handled.

I like your advise on rethrowing a custom exception and catching it higher up.  I'll leave this as is in this PR, but might work on an enhancement in a separate PR later.  Thanks for the tip.